src/lib/Libpython/pbs_python_external.c

```diff
@@ -278,6 +238,28 @@ pbs_python_ext_start_interpreter(struct python_interpreter_data *interp_data)
 	}
 
 	interp_data->pbs_python_types_loaded = 1; /* just in case */
+
+#if !defined(PBS_PYTHON)
+	PyObject *m, *d, *f, *handler, *sigint;
+    m = PyImport_ImportModule("signal");
+    d = PyModule_GetDict(m);
+    f = PyDict_GetItemString(d, "signal");
+    handler = PyDict_GetItemString(d, "default_int_handler");
+    sigint = PyDict_GetItemString(d, "SIGINT");
+    if (f && PyCallable_Check(f)) {
+        PyObject_CallFunctionObjArgs(f, sigint, handler, NULL);
+    } else {
+        log_err(-1, __func__, "could not set up python signal.default_int_handler");
+        goto ERROR_EXIT;
+    }
+    Py_XDECREF(m);
+    Py_XDECREF(d);
+    Py_XDECREF(f);
+    Py_XDECREF(sigint);
+    Py_XDECREF(handler);
+    log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_INFO, interp_data->daemon_name, "successfully set up signal.default_int_handler");
+#endif
+
```

Comment:
Could this also be the reason that SIGALRM isn't getting through? The Timeout class in the cgroups hook seems to be broken since we switched to Python 3. @alexis-cousein will also be interested.

Reply:
Yes this could be the reason. Python 3 no longer sets up `default_int_handler` when it starts like python 2 did.

Reply:
Should we be doing this for additional signals, besides SIGINT? Or is there a way to cope with this in the hook itself?

Reply:
We are doing this for SIGINT because we rely on `default_int_handler`'s behavior (raising a KeyboardInterrupt exception) for hook alarms. 
For all the other signals that each hook want to handler, the hook itself should import the `signal` module and register and custom handler function using `signal.signal()`.